src/rngs/adapter/read.rs

```diff
@@ -131,6 +124,13 @@ mod test {
 
         let mut rng = ReadRng::new(&v[..]);
 
-        assert!(rng.try_fill_bytes(&mut w).err().unwrap().kind == ErrorKind::Unavailable);
+        let err_kind = rng.try_fill_bytes(&mut w)
+            .err()
+            .unwrap()
+            .cause()
+            .downcast_ref()
+            .map(|e: &io::Error| e.kind())
+            .unwrap();
+        assert_eq!(err_kind, io::ErrorKind::UnexpectedEof);
```

Comment:
This is more specific than previously since all errors used to map to `Unavailable`. (Which is fine; just commenting.)